clients/python/test/e2e/test_notebooks.py

```diff
@@ -0,0 +1,33 @@
+import papermill as pm
+from pathlib import Path
+from tempfile import TemporaryDirectory
+import shutil
+import pytest
+from typing import List, Dict, Any
+import sys
+import osparc
+
+docs_dir:Path = Path(__file__).parent.parent.parent / 'docs'
+all_notebooks:List[Path] = list(docs_dir.glob("*.ipynb"))
+
+def test_notebook_config():
+    """ Checks the jupyter environment is configured correctly
+    """
+    config_test_nb:Path = Path(__file__).parent / 'data' / 'config_test.ipynb'
+    assert config_test_nb.is_file()
+    test_run_notebooks(config_test_nb,{'expected_python_bin':sys.executable, 'expected_osparc_version':str(osparc.__version__), 'expected_osparc_file':osparc.__file__})
+    assert len(all_notebooks) > 0, f'Did not find any notebooks in {docs_dir}'
+
+
+@pytest.mark.parametrize('notebook,params',list(zip(all_notebooks,[{}]*len(all_notebooks))))
+def test_run_notebooks(notebook:Path, params:Dict[str,Any]):
+    """Run all notebooks in the documentation
+    """
+    print(f"Running {notebook.name} with parameters {params}")
+    assert notebook.is_file(), f'{notebook.name} is not a file (full path: {notebook.resolve()})'
+    with TemporaryDirectory() as tmp_dir:
```

Comment:
you can have a fixture for the tmpdir as 
```python
@pytest.mark.parametrize('notebook,params',list(zip(all_notebooks,[{}]*len(all_notebooks))))
def test_run_notebooks(notebook:Path, params:Dict[str,Any], tmp_path: Path):
    tmp_nb = Path(tmp_path) / notebook.name
    ...
```

SEE https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html

Reply:
Good point. I have switched to the fixture instead.